collections/ansible_collections/cloudkit/service/roles/manage_agents/tasks/import_agents.yaml

```diff
@@ -0,0 +1,184 @@
+- name: Display node names that will be added as agents
+  ansible.builtin.debug:
+    var: manage_agents_node_names
+
+- name: Get list of existing agents
+  kubernetes.core.k8s_info:
+    kind: Agent
+    api_version: agent-install.openshift.io/v1beta1
+    namespace: "{{ manage_agents_namespace }}"
+  register: initial_agents
+
+- name: Get node informations
+  openstack.cloud.baremetal_node_info:
+    name: "{{ node_name }}"
+  register: node_infos
+  loop: "{{ manage_agents_node_names }}"
+  loop_control:
+    loop_var: node_name
+
+- name: Extract node informations
+  ansible.builtin.set_fact:
+    node_info_list: "{{ node_infos.results | map(attribute='baremetal_nodes') | map('first') | list }}"
+
+- name: Get MAC addresses of nodes
+  openstack.cloud.baremetal_port_info:
+    node: "{{ node_name }}"
+  register: ports_list
+  loop: "{{ manage_agents_node_names }}"
+  loop_control:
+    loop_var: node_name
+
+- name: Extract MAC addresses
+  ansible.builtin.set_fact:
+    node_addresses_list: >-
+      {{
+        ports_list.results |
+        map(attribute='ports') |
+        map('map', attribute='address') |
+        map('list') |
+        list
+      }}
+
+- name: Set filtered lists of only nodes that need importing
+  ansible.builtin.set_fact:
+    filtered_node_info: "{{ filtered_node_info | default([]) + [node_info_list[node_idx]] }}"
+    filtered_node_addresses: "{{ filtered_node_addresses | default([]) + [node_addresses_list[node_idx]] }}"
+  loop: "{{ range(manage_agents_node_names | length) | list }}"
+  loop_control:
+    loop_var: node_idx
+  when: not node_addresses_list[node_idx] | massopencloud.esi.match_agent_addresses(initial_agents.resources)
+
+- name: Get InfraEnv information
+  kubernetes.core.k8s_info:
+    kind: InfraEnv
+    api_version: agent-install.openshift.io/v1beta1
+    namespace: "{{ manage_agents_namespace }}"
+    name: "{{ manage_agents_infraenv_name }}"
+  register: infraenv
+
+- name: Set discovery_url
+  ansible.builtin.set_fact:
+    discovery_url: "{{ infraenv.resources[0].status.isoDownloadURL }}"
+  when: not discovery_url|default(false)
+
+- name: Set node to provisioning network
+  ansible.builtin.include_role:
+    name: massopencloud.esi.l2
+    tasks_from: set_networks_for_node
+  vars:
+    networks:
+      - "{{ manage_agents_provisioning_network_name }}"
+    node: "{{ node_info.name }}"
+  loop: "{{ filtered_node_info }}"
+  loop_control:
+    loop_var: node_info
+
+- name: Set deploy interface, set boot image url, and deploy node
+  openstack.cloud.baremetal_node_action:
+    name: "{{ node_info.name }}"
+    instance_info:
+      deploy_interface: ramdisk
+      boot_iso: "{{ discovery_url }}"
+    deploy: true
+  loop: "{{ filtered_node_info }}"
+  loop_control:
+    loop_var: node_info
```

Comment:
`Set node to provisioning network` and this task could be combined into a single `massopencloud.esi` node deployment role; I would suggest something under `massopencloud/esi/roles/node`